### PR TITLE
Previews

### PR DESCRIPTION
**Pull request by AndrewBrehmAtDialexa**
Added previews for all Views. Also added mock items to allow viewing of various states within views. Utilized mocks so that navigation can be implemented for increased development

### CODE REVIEW

**Review by leemccormick — APPROVED** [review]

File: CommonUI/Sources/CommonUI/Buttons/ButtonStyle.swift [hunk 1]
```diff
@@ -25,3 +25,16 @@ public struct LightGreenButton: ButtonStyle {
             .clipShape(Capsule())
     }
 }
+
+#Preview(body: {
```
Comment (hain2005) [inline CommonUI/Sources/CommonUI/Buttons/ButtonStyle.swift]: You don't need `body` specify.  
#Preview {
       VStack {
       .....
       }
 }

File: SearchMicro/Sources/SearchMicro/SearchNavigation/SearchNavigationViewModel.swift [hunk 2]
```diff
@@ -3,7 +3,11 @@ import FeatureView
 
 class SearchNavigationViewModel {
     // MARK: - View Builders
-    
+
+    func createSearchView() -> SearchView {
+        return SearchView()
```
Comment (hain2005) [inline SearchMicro/Sources/SearchMicro/SearchNavigation/SearchNavigationViewModel.swift]: Don't need `return` here.

File: SearchMicro/Sources/SearchMicro/SearchNavigation/SearchNavigationViewModel.swift [hunk 3]
```diff
@@ -3,7 +3,11 @@ import FeatureView
...
+    }
+
     func createMovieDetailsView(movie: Movie) -> MovieDetailsView {
         return MovieDetailsView(movie: movie)
```
Comment (hain2005) [inline SearchMicro/Sources/SearchMicro/SearchNavigation/SearchNavigationViewModel.swift]: Here too, missed it last time.

**Comment by hain2005** [comment]
You are using #Previews macro instead from your examples you showed (HomeView_Previews).  Is it because too much work to add for all micros?

File: SearchMicro/Sources/SearchMicro/SearchNavigation/SearchNavigationViewModel.swift (hunk 2 above)
Comment (AndrewBrehmAtDialexa) [reply SearchMicro/Sources/SearchMicro/SearchNavigation/SearchNavigationViewModel.swift]: oooh good catch!
updated

**Review by hain2005 — APPROVED** [review]

**Pull request merged**